client/cli/src/main.rs

```diff
@@ -181,6 +203,22 @@ fn print_help(client_info: &str, commands: &[std::sync::Arc<dyn Command>]) {
     println!("\n");
 }
 
+/// Retrieve a waypoint given the URL.
+fn retrieve_waypoint(url_str: &str) -> anyhow::Result<Waypoint> {
+    let mut data = Vec::new(); // waypoint bytes downloaded from the given URL
+    let mut handle = Easy::new();
```

Comment:
@davidiw did you evaluate this client library? 😆

Reply:
I would recommend we just use ureq throughout our codebase unless there's a need for an async web client.

Reply:
Shared offline: https://github.com/libra/libra/blob/master/secure/storage/vault/src/lib.rs#L87

Reply:
ureq doesn't support system-wide proxy ,  and some client needs to run on tupperware and devserver with fwdproxy configured.

Reply:
I think we could work with them to add that feature. HTTP Proxy is pretty trivial: https://en.wikipedia.org/wiki/Proxy_server#Implementations_of_proxies

Reply:
The actual proxy protocol support is relatively easy. But handle all corner cases where http_proxy https_proxy and no_proxy are defined are not. Which is also why most implementation doesn’t work out of the box . I am not saying it is impossible to do, but it is well beyond the scope of this change. 

After all, reqwest has already been added to CLI dependency tree to solve this exact issue, so there are no new additional deps to be add.

Reply:
reqwest pulls in so many dependencies into the client code base. I would prefer we challenge ourselves a little bit and ask the community for help rather than bloating the dependency tree. If, when closer to launch, we this hasn't been resolved, I am happy to support changing to a different library that supports our needs.

Reply:
https://github.com/algesten/ureq/issues/39